### DISCUSSION TITLE
Is ImageMagick vulnarable to CVE-2023-4863? (libwebp heap overflow leading to RCE)

### DISCUSSION THREAD

**Discussion by dedene** [ImageMagick/ImageMagick#6746] [General]
[CVE-2023-4863](https://nvd.nist.gov/vuln/detail/CVE-2023-4863) was modified when it became clear the bug is not limited to Chromium, but impacting any software using libwebp. 

Therefore, I'm wondering: is ImageMagick using this library for webp processing and thus vulnerable to Remote Code Execution using a specially crafted webp image?

More info [here](https://stackdiary.com/critical-vulnerability-in-webp-codec-cve-2023-4863/).

**Comment by dlemstra** [comment]
I suspect that our code can hit that same path when someone builds ImageMagick with a version of libwebp that includes that vulnerability.

**Comment by urban-warrior** [comment]
Assuming your release of ImageMagick includes WebP support, remediation is to either upgrade your release of the `libwebp` delegate library, build ImageMagick without WebP support, or add a WebP denial security policy to your current IM build:

```
<!--  These image types are security risks on read, but write is fine  -->
<policy xmlns="" domain="module" rights="write" pattern="{MSL,MVG,PS,SVG,URL,WEBP,XPS}"/>
```

See https://imagemagick.org/script/security-policy.php for details about the ImageMagick security policy.

**Comment by marklundin** [comment]
It seems there is a [patch for this](https://github.com/webmproject/libwebp/commit/902bc9190331343b2017211debcec8d2ab87e17a) in release [v1.3.2](https://github.com/webmproject/libwebp/releases/tag/v1.3.2) of libwebp.

**Comment by svenclaesson** [comment]
According to https://github.com/ImageMagick/ImageMagick/discussions/6664
This is fixed in 7.1.1-17
